oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/query/async/ElasticResultRowAsyncIterator.java

```diff
@@ -323,7 +338,8 @@ public void onSuccess(SearchResponse<ObjectNode> searchResponse) {
         public void onFailure(Throwable t) {
             metricHandler.measureFailedQuery(indexNode.getDefinition().getIndexPath(),
                     System.currentTimeMillis() - searchStartTime);
-            LOG.error("Error retrieving data from Elastic: closing scanner, notifying listeners", t);
+            errorRef.set(t);
```

Comment:
This is not checking if errorRef is already set. In that case, the exception stored there will be lost. If we assume that onFailure is called only once, then it's fine (although we may want to log a warning in case errorRef is already set, `getAndSet()` could once again be useful here). But if the assumption is that it can be called multiple times, then this can lead to losing exceptions.

Reply:
Done.